src/boost_histogram/_internal/hist.py

```diff
@@ -854,13 +854,15 @@ def __getitem__(self: H, index: IndexingExpr) -> H | float | Accumulator:
             if ind != slice(None):
                 merge = 1
                 if ind.step is not None:
-                    if hasattr(ind.step, "factor"):
+                    if getattr(ind.step, "factor", None) is not None:
                         merge = ind.step.factor
                     elif callable(ind.step):
                         if ind.step is sum:
                             integrations.add(i)
+                        elif getattr(ind.step, "groups", None) is not None:
```

Comment:
This should call `ind.step`, not check for the `groups` attr. The rebinner needs to be able to compute this on the fly, with the axis as an input.